Book-A-Desk.Api/InputParser.fs

```diff
@@ -25,4 +25,12 @@ module InputParser =
         let wasParsed, officeIdGuid = Guid.TryParse reference
         let officeId = officeIdGuid |> OfficeId        
         if wasParsed then Ok officeId
-        else Error ()
\ No newline at end of file
+        else Error ()
+
+    let parseOfficeIdFromContext (context : HttpContext) =
+        context.TryGetQueryStringValue "office"
+        |> Option.bind (fun (office: string) ->
```

Comment:
I think this code is a little weird since we are mixing both options and errors when trying to reuse the existing parseOfficeId. I think a refactor will help here:
1. Change parseOfficeId to return an Option instead of an error.
2. In parseOfficeIdFromContext, you can call parseOfficeId directly
3. In the code that currently calls parseOfficeId, you can handle the Option instead of a Result. You probably will have to map your Option<OfficeId> to a Result<OfficeId, String> in the existing callers.